flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -210,7 +211,12 @@ public String getDescription() throws ProgramInvocationException {
      * local execution by default.
      */
     public void invokeInteractiveModeForExecution() throws ProgramInvocationException {
-        callMainMethod(mainClass, args);
+        FlinkUserSecurityManager.monitorSystemExitForCurrentThread();
```

Comment:
There won't be a `FlinkUserSecurityManager` configured if the `CliFrontend` is used.

Reply:
IIUC, it's okay not to have `FlinkUserSecurityManager` in `CliFrontend`, as our focus is more on JM like crashing Flink cluster itself. Do you want me to also apply this monitoring to CliFrontend? Currently, I have not done it.

Reply:
Ah, you have added this here because in some execution modes, the JobManager process is running Flink's main methods.
Makes sense. Consider my comment resolved.